youtube_dl/options.py

```diff
@@ -177,6 +177,11 @@ def _hide_login_info(opts):
         default=None, metavar='URL',
         help='Use the specified HTTP/HTTPS proxy. Pass in an empty string (--proxy "") for direct connection')
     network.add_option(
+        '--alternative-proxy', dest='alternative_proxy',
```

Comment:
Instead of doing that on each extractor, I think it should be done for all of them (for example in the `InfoExtractor.extract` method). It would be more consistent.